nexus/src/authz/api_resources.rs

```diff
@@ -0,0 +1,303 @@
+
+impl Fleet {
+    /// Returns an authz resource representing a child Organization
+    pub fn organization(&self, organization_id: Uuid) -> Organization {
+        Organization { organization_id }
+    }
+
+    /// Returns an authz resource representing some other kind of child (e.g.,
+    /// a built-in user, built-in role, etc. -- but _not_ an Organization)
+    ///
+    /// Aside from Organizations (which you create with [`Fleet::organization()`
+    /// instead), all instances of all types of Fleet children are treated
+    /// interchangeably by the authz subsystem.  That's because we do not
+    /// currently support assigning roles to these resources, so all that
+    /// matters for authz is that they are a child of the Fleet.
+    pub fn child_generic(&self) -> FleetChild {
+        FleetChild {}
+    }
+}
+
+impl oso::PolarClass for Fleet {
+    fn get_polar_class_builder() -> oso::ClassBuilder<Self> {
+        oso::Class::builder().with_equality_check().add_method(
+            "has_role",
+            |_: &Fleet, actor: AuthenticatedActor, role: String| {
+                actor.has_role_resource(ResourceType::Fleet, *FLEET_ID, &role)
+            },
+        )
+    }
+}
+
+impl AuthzApiResource for Fleet {
+    fn db_resource(&self) -> Option<(ResourceType, Uuid)> {
+        Some((ResourceType::Fleet, *FLEET_ID))
+    }
+
+    fn parent(&self) -> Option<Box<dyn AuthzResource>> {
+        None
+    }
+}
+
+/// Represents any child of a Fleet _except_ for Organizations for authz
+/// purposes
+///
+/// This includes [`crate::db::model::UserBuiltin`], similar authz-related
+/// resources, and potentially future resources describing hardware.
+///
+/// This object is used for authorization checks on such resources by passing
+/// this as the `resource` argument to
+/// [`crate::context::OpContext::authorize()`].  You construct one of these
+/// using [`Fleet::child_generic()`].
+// We do not currently store any state here because we don't do anything with
+// it, so if callers provided it, we'd have no way to test that it was correct.
+// If we wind up supporting attaching roles to these, then we could add a
+// resource type and id like we have with ProjectChild.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct FleetChild {}
+
+impl oso::PolarClass for FleetChild {
+    fn get_polar_class_builder() -> oso::ClassBuilder<Self> {
+        oso::Class::builder()
+            .with_equality_check()
+            .add_method(
+                "has_role",
+                /* Roles are not supported on FleetChilds today. */
+                |_: &FleetChild, _: AuthenticatedActor, _: String| false,
+            )
+            .add_attribute_getter("fleet", |_: &FleetChild| Fleet {})
+    }
+}
+
+impl AuthzApiResource for FleetChild {
+    fn db_resource(&self) -> Option<(ResourceType, Uuid)> {
+        None
+    }
+
+    fn parent(&self) -> Option<Box<dyn AuthzResource>> {
+        Some(Box::new(Fleet {}))
```

Comment:
Same thing here. Any equality check would of course still pass, but that won't be true if we change the `Fleet` type to include state. It might also be good documentation to identify this as `FLEET`. Again, assuming that's actually what we want, I'm still getting my bearings in this code.

Reply:
Fixed in 00ac778eadf087dc0be1d8d652dbeee55521189a.